refactor(Account Management)

lomamech opened the pull request:
- Upgrate Account management module us Dropdown Menus
- Remove bhRenderDropDown
- Improve chart handlebars
- Add function download in AccountGridService

closes #1914

lomamech committed:
refactor(Account Management)
- Upgrate Account management module us Dropdown Menus
- Remove bhRenderDropDown
- Improve chart handlebars
- Add function download in AccountGridService

jniles commented on client/src/modules/accounts/AccountGrid.js:
```diff
@@ -48,6 +48,19 @@ function AccountGridService(AccountStore, Accounts, Store, Notify) {
   };
 
+
+  AccountGrid.prototype.download = function download(type, filters) {
```
Cool :+1:

jniles commented on client/src/modules/accounts/accounts.html:
```diff
@@ -1,37 +1,51 @@
-<div class="flex-header static">
+<div class="flex-header">
```
:dancer:

jniles commented on client/src/modules/accounts/accounts.html:
```diff
@@ -1,37 +1,51 @@
 
-   <div class="toolbar">
+    <div class="toolbar">
+      <fieldset ng-disabled="AccountsCtrl.transactions.isEditing()">
```
I don't think this is right - `AccountsCtrl` shouldn't have a `transactions` property.

jniles commented on client/src/modules/accounts/accounts.html:
```diff
@@ -1,37 +1,51 @@
-        options="AccountsCtrl.parameter">
-      </bh-pdf-print>
+              <li role="menuitem">
+                <a ng-href="/reports/finance/accounts/chart/?{{ AccountsCtrl.Accounts.download('csv') }}" download="{{ 'ACCOUNT.TITLE' | translate }}">
```
@lomamech, while you are at it, @jeremielodi recently added downloading as Excel (https://github.com/IMA-WorldHealth/bhima-2.X/pull/2106).  Could you add that feature into this list as well?

jniles commented on client/src/modules/accounts/accounts.html:
```diff
@@ -1,37 +1,51 @@
-        report-url="/reports/finance/accounts/chart"
-        report-options="AccountsCtrl.parameter">
-      </bh-renderer-dropdown>
+        <div class="toolbar-item">
```
Nice!

jniles commented on server/controllers/finance/reports/accounts/chart.handlebars:
```diff
@@ -23,7 +23,7 @@
     <tbody>
       {{#each accounts}}
-        {{#if this.is_title_account}}
+        {{#if this.is_title}}
```
:dancer:

jniles commented on server/controllers/finance/reports/accounts/index.js:
```diff
@@ -24,6 +24,7 @@ function chart(req, res, next) {
     csvKey : 'accounts',
     filename : 'REPORT.CHART_OF_ACCOUNTS',
     orientation : 'landscape',
+    footerRight : '[page] / [toPage]',
```
Nice!

lomamech committed:
perf(Account Management)
- Remove Unecessary balise fieldset with the option Editing
- Add the option downloading as CSS
